### PR TITLE
[SPARK-50069][SQL] Integrate `_LEGACY_ERROR_TEMP_0028` into `UNSUPPORTED_FROM_TO_EXPRESSION`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to Integrate `_LEGACY_ERROR_TEMP_0028` into `UNSUPPORTED_FROM_TO_EXPRESSION`

### Why are the changes needed?

To improve the error message by assigning proper error condition and SQLSTATE

### Does this PR introduce _any_ user-facing change?

No, only user-facing error message improved

### How was this patch tested?

Updated the existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala [hunk 1]
```diff
@@ -126,7 +133,14 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
...
       val end = DayTimeIntervalType.stringToField(endStr)
       if (end <= start) {
-        throw QueryParsingErrors.fromToIntervalUnsupportedError(startStr, endStr, ctx)
+        val intervalInput = ctx.getText()
+        val pattern = "'([^']*)'".r
+
+        val input = pattern.findFirstMatchIn(intervalInput) match {
+          case Some(m) => m.group(1)
+          case None => ""
+        }
```
Comment (MaxGekk) [inline sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala:142]: could you put the parsing error code into `fromToIntervalUnsupportedError`.

Comment (itholic) [reply sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala]: Updated. Thanks!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Comment by itholic** [comment]
Thanks @MaxGekk for the review

**Pull request closed without merging**